WalletWasabi/Crypto/Randomness/WasabiRandom.cs

```diff
@@ -23,7 +23,7 @@ public virtual byte[] GetBytes(int length)
 		public abstract int GetInt(int fromInclusive, int toExclusive);
 
 		public double NextDouble() =>
-			Math.Abs(BitConverter.ToDouble(GetBytes(sizeof(double)), 0) / double.MaxValue);
+			GetInt(0, int.MaxValue) / (double)int.MaxValue;
```

Comment:
just sample int range for the time duration and convert to timespan?

Reply:
This method only returns a double in the range [0, 1) which is used to sample the time frame.

Reply:
I understand that, but I still think it's superfluous. why not just do `new TimeSpan(rnd.GetInt(0, totalTimeSpan.Ticks))`? it's much more direct and easier to reason about.

That way we don't have to think about the sampling bias *or* any sort of loss because of floating point arithmetic, in particular by normalizing to int.MaxValue and dividing. after thinking about it, I don't see a problem with doing it this way, but it took me a fair bit of thinking, reminding myself that doubles are able to represent integer arithmetic accurately up to about 2^53... but this is just converting an int to a double in order to convert it back to an int at the end, that's a lot of mental effort for something that should be straightforward and easy to audit. in particular whenever floats are involved in anything sensitive like this my brain slows down 10x because they are a huge red flag.

Reply:
ah crap, I don't know why i keep reading `int` as `int64` instead of `int32`.

anyway, i would still prefer an implementation of `GetLong` over an implementation of `NextDouble`

that said, for the time spans we care about, ints are perfectly fine, there are only ~2^22 milliseconds in an hour

Reply:
using ms now. done.